docs/csharp/whats-new/csharp-version-history.md

```diff
@@ -0,0 +1,124 @@
+
+A more nuanced view examines expression tress, lamba expressions and anonymous types as the foundation upon which LINQ is constructed.  But, in either case, C# 3 presented with a fairly revolutionary concept.  C# 3 had begun to lay the groundwork for turning C# into a hybrid OO-functional language.
+
+Specifically, you could now write SQL-style, declarative queries to perform operations on collections, among other things.  Instead of writing a for loop to compute the average of a list of integers, you could now do that as simply as list.Average().  The combination of query expressions and extension methods made it look as though that list of ints had gotten a whole lot smarter.
+
+It took a little while for people to really grasp and integrate the concept, but they gradually did.  And now, years later, code is much more concise, simple, and functional.
+
+## C# Version 4
+
+C# Version 4 would have had a difficult time living up to the groundbreaking status of version 3.  With version 3, C# had moved the language firmly out from the shadow of Java and into prominence.  The language was quickly becoming elegant
+
+The next version did introduce some cool stuff, though.
+
+- [Dynamic binding](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/dynamic)
+- [Named/optional arguments](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments)
+- [Generic covariant and contravariant](https://docs.microsoft.com/en-us/dotnet/standard/generics/covariance-and-contravariance)
+- [Embedded interop types](https://stackoverflow.com/questions/20514240/whats-the-difference-setting-embed-interop-types-true-and-false-in-visual-studi)
+
+Embedded interop types alleviated a deployment pain.  Generic covariance and contravariance give you a lot of power, but they&#39;re a bit academic and probably most appreciated by framework and library authors.  Named and optional parameters let you eliminate a lot of method overloads and provide convenience.  But none of those are exactly paradigm altering.
+
+The major feature was the introduction of the dynamic keyword.  The `dynamic` introduced into C# Version 4 the ability to override the compiler on compile time typing.  By using the dynamic keyword, you can create constructs similar to dynamically typed languages like JavaScript.  You can create a dynamic x = &quot;a string&quot; and then add six to it, leaving it up to the runtime to sort out what on should happen next.
+
+This gives you the potential for errors but also great power within the language.
+
+## C# Version 5
+
+C# Version 5 wasa very focused version of the language.  Nearly all of their effort for that version into another pretty groundbreaking language concept.  Here is the major features list.
```

Comment:
wasa -> was a 

remove pretty

Second sentence is missing a verb. It's probably went?